packages/angular-ivy/scripts/prepack.ts

```diff
@@ -15,12 +15,12 @@ const pkgJson: PackageJson = JSON.parse(fs.readFileSync(pkjJsonPath).toString())
 // This is necessary for Angular 17+ compatibility when SSR is configured which switches dev mode to using Vite.
 // Deleting "main" and adding "type": "module" will direct Vite to
 // use the fesm2015 bundle instead of the UMD bundle.
-delete pkgJson.main;
+pkgJson.main = undefined;
```

Comment:
I think we intentionally `delete` so the key is removed from the JSON that is written to disk.

Reply:
I don't think that holds, since on serialization undefined values are omitted by default?

```
> JSON.stringify({ hello: undefined })
'{}'
```